kyuubi-server/src/main/scala/org/apache/kyuubi/server/mysql/MySQLCommandHandler.scala

```diff
@@ -0,0 +1,248 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.server.mysql
+
+import java.util.concurrent.{ConcurrentHashMap, ThreadPoolExecutor}
+import java.util.concurrent.atomic.AtomicInteger
+
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutor, Future}
+import scala.util.{Failure, Success}
+
+import io.netty.channel.{ChannelHandlerContext, SimpleChannelInboundHandler}
+import org.apache.hive.service.rpc.thrift.TProtocolVersion
+
+import org.apache.kyuubi.{KyuubiSQLException, Logging}
+import org.apache.kyuubi.operation.FetchOrientation
+import org.apache.kyuubi.operation.OperationState._
+import org.apache.kyuubi.server.mysql.MySQLCommandHandler._
+import org.apache.kyuubi.server.mysql.constant.{MySQLDataType, MySQLServerDefines}
+import org.apache.kyuubi.server.mysql.constant.MySQLCtxAttrKey._
+import org.apache.kyuubi.service.BackendService
+import org.apache.kyuubi.session.SessionHandle
+
+object MySQLCommandHandler {
+  val connIdCounter = new AtomicInteger
+  val connIdToSessHandle = new ConcurrentHashMap[Int, SessionHandle]
+}
+
+class MySQLCommandHandler(be: BackendService, execPool: ThreadPoolExecutor)
+  extends SimpleChannelInboundHandler[MySQLCommandPacket] with Logging {
+
+  private implicit val ec: ExecutionContextExecutor = ExecutionContext.fromExecutor(execPool)
+
+  @volatile private var closed: Boolean = false
+
+  override def channelInactive(ctx: ChannelHandlerContext): Unit = {
+    closeSession(ctx)
+    super.channelInactive(ctx)
+  }
+
+  // handle process exception, generally should send error packet
+  override def exceptionCaught(ctx: ChannelHandlerContext, cause: Throwable): Unit = {
+    val connectionId = ctx.channel.attr(CONNECTION_ID).get
+    val errPacket = MySQLErrPacket(cause)
+    error(s"Connection: $connectionId, $errPacket")
+    if (ctx.channel.isActive) {
+      ctx.writeAndFlush(errPacket)
+    } else {
+      warn(s"Ignore error packet for inactivated connection: $connectionId")
+    }
+  }
+
+  override def channelRead0(ctx: ChannelHandlerContext, packet: MySQLCommandPacket): Unit = {
+    Future {
```

Comment:
Process all command in `ec: ExecutionContextExecutor`, do not block netty worker event loop.